test/test_atacac/test__utils.py

```diff
@@ -0,0 +1,191 @@
+    mock_instance.list.return_value = {"results": result_assets}
+
+    try:
+        retval = _utils.tower_list(asset_type, query)
+    except _utils.Error as e:
+        assert str(e) == error
+        assert e.error_code == 1
+    else:
+        mock_get_resource.assert_called_once()
+        mock_get_resource.assert_called_with(asset_type)
+
+        mock_instance.list.assert_called_once()
+        mock_instance.list.assert_called_with(all_pages=True, query=("label", 1))
+
+        assert retval == result_assets
+
+
+@pytest.mark.parametrize(
+    "error, asset_type, asset_name",
+    [
+        pytest.param(None, "user", "file", id="user"),
+        pytest.param(None, "organization", "file", id="organization"),
+        pytest.param(None, "team", "file", id="team"),
+        pytest.param(None, "credential_type", "file", id="credential_type"),
+        pytest.param(None, "credential", "file", id="credential"),
+        pytest.param(None, "notification_template", "file", id="notification_template"),
+        pytest.param(None, "inventory_script", "file", id="inventory_script"),
+        pytest.param(None, "project", "file", id="project"),
+        pytest.param(None, "inventory", "file", id="inventory"),
+        pytest.param(None, "job_template", "file", id="job_template"),
+        pytest.param(None, "workflow", "file", id="workflow"),
+        pytest.param("Unsupported asset type 'ABC123XYZ'!",
+                     "ABC123XYZ", "file", id="unsupported asset type")
+    ]
+)
+@mock.patch("atacac._utils.Receiver")
+def test_tower_receive(mock_Receiver, error, asset_type, asset_name):
+    dictionary = {"dictA": {"key_1": "value_1"}, "dictB": {"key_2": "value_2"}}
+
+    mock_instance = mock_Receiver.return_value
+    mock_instance.export_assets.return_value = dictionary
+
+    try:
+        assert _utils.tower_receive(asset_type, asset_name) == dictionary
+    except _utils.Error as e:
+        assert str(e) == error
+        assert e.error_code == 1
+    else:
+        mock_Receiver.assert_called_once()
+
+        mock_instance.export_assets.assert_called_once()
+        mock_instance.export_assets.assert_called_with(all=False, asset_input={asset_type: [asset_name]})
+
+
+@pytest.mark.parametrize(
+    "assets",
+    [
+        pytest.param("foo", id="file"),
+        pytest.param(["foo", "bar"], id="file list"),
+    ]
+)
+@mock.patch("atacac._utils.Sender")
+def test_tower_send(mock_Sender, assets):
+    mock_instance = mock_Sender.return_value
+
+    _utils.tower_send(assets)
+
+    mock_Sender.assert_called_once()
+    mock_Sender.assert_called_with(False)
+
+    mock_instance.send.assert_called_once()
+    mock_instance.send.assert_called_with(
+        assets if isinstance(assets, list) else [assets], None, None, "default")
+
+
+@pytest.mark.parametrize(
+    "valid",
+    [
+        pytest.param(True, id="valid file path"),
+        pytest.param(False, id="invalid file path")
+    ]
+)
+def test_load_asset(valid):
+    if valid:
```

Comment:
You could split this test case to 2 functions and get rid of this `if`. It is better to keep tests as simple as possible.